hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,822 @@
+                E = component.centre.value
+                component.sigma.twin_function = _get_sigma(
+                    E, E_ref, self.units_factor)
+                component.sigma.twin_inverse_function = _get_sigma(
+                    E_ref, E, self.units_factor)
+                component.sigma.twin = component_ref.sigma
+
+    def _set_energy_resolution(self, xray_lines, ref=None):
+        """
+        Adjust the width of all lines and set the fitted energy resolution
+        to the spectrum
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        ref: None
+            dummy args, to work like other set_..._energy
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        energy_Mn_Ka, FWHM_MnKa_old = self.spectrum._get_line_energy('Mn_Ka',
+                                                                     'auto')
+        FWHM_MnKa_old *= 1000. / self.units_factor
+        get_sigma_Mn_Ka = _get_sigma(
+            energy_Mn_Ka, self[xray_lines[0]].centre.value, self.units_factor)
+        FWHM_MnKa = get_sigma_Mn_Ka(self[xray_lines[0]].sigma.value
+                                    ) * 1000. / self.units_factor * 2.355
+        if FWHM_MnKa < 110:
+            print "FWHM_MnKa of " + str(FWHM_MnKa) + " smaller than " + \
+                "physically possible"
+        else:
+            self.spectrum.set_microscope_parameters(
+                energy_resolution_MnKa=FWHM_MnKa)
+            print("Energy resolution (FWHM at Mn Ka) changed from "
+                  + "%lf to %lf eV" % (FWHM_MnKa_old, FWHM_MnKa))
+            for component in self:
+                if component.isbackground is False:
+                    line_energy, line_FWHM = self.spectrum._get_line_energy(
+                        component.name, FWHM_MnKa='auto')
+                    component.sigma.value = line_FWHM / 2.355
+
+    def _twin_xray_lines_scale(self, xray_lines):
+        """
+        Twin the scale of the peaks
+
+        Parameters
+        ----------
+        xray_lines: list of str or 'all_alpha'
+            The Xray lines. If 'all_alpha', fit all using all alpha lines
+        """
+        if xray_lines == 'all_alpha':
+            xray_lines = [compo.name for compo in self.xray_lines]
+        ax = self.spectrum.axes_manager[-1]
+        ref = []
+        for i, xray_line in enumerate(xray_lines):
+            component = self[xray_line]
+            if i == 0:
+                component_ref = component
+                component_ref.centre.free = True
+                E_ref = component_ref.centre.value
+                ref.append(E_ref)
+            else:
+                component.centre.free = True
+                E = component.centre.value
+                fact = float(ax.value2index(E)) / ax.value2index(E_ref)
```

Comment:
I think the best way to figure this out is to create a test that adds both a scaling and an offset, and see if order is important or not, then keep the test(s) for the order that works, and document accordingly.